pegasus/helpers/section_api_helpers.rb

```diff
@@ -21,23 +21,29 @@ def self.fetch_user_students(user_id)
   def self.create(params)
     name = !params[:name].to_s.empty? ? params[:name].to_s : 'New Student'
     gender = valid_gender?(params[:gender]) ? params[:gender] : nil
+    provider = valid_provider?(params[:provider]) ? params[:provider] : 'sponsored'
     birthday = age_to_birthday(params[:age]) ?
       age_to_birthday(params[:age]) : params[:birthday]
 
     created_at = DateTime.now
 
-    row = Dashboard.db[:users].insert(
+    data =
       {
         name: name,
         user_type: 'student',
-        provider: 'sponsored',
+        provider: provider,
         gender: gender,
         birthday: birthday,
         created_at: created_at,
         updated_at: created_at,
         username: UserHelpers.generate_username(Dashboard.db[:users], name)
-      }.merge(random_secrets)
-    )
+      }
+    if provider == 'sponsored'
+      data.merge!(random_secrets)
+    else
+      data[uid] = params[:uid].to_i
```

Comment:
The DB shows `users.uid` as being a `VARCHAR(255)`. Given this, is the cast to an integer appropriate?
```
SHOW CREATE TABLE users;
...
  `uid` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
...
```